plugins/cue_points/quiz/lib/api/KalturaAnswerCuePoint.php

```diff
@@ -88,31 +80,63 @@ public function doFromObject($dbObject, KalturaDetachedResponseProfile $response
 		parent::doFromObject($dbObject, $responseProfile);
 
 		$dbEntry = entryPeer::retrieveByPK($dbObject->getEntryId());
-		if ( !kEntitlementUtils::isEntitledForEditEntry($dbEntry) ) {
+		if ( !kEntitlementUtils::isEntitledForEditEntry($dbEntry) )
+		{
 			/**
 			 * @var kQuiz $kQuiz
 			 */
 			$kQuiz = QuizPlugin::validateAndGetQuiz( $dbEntry );
 
+			//Find the corresponding question cue point
+			$this->validateParentId();
+			$dbQuestionCuePoint = CuePointPeer::retrieveByPK($this->parentId);
+
+			$kQuestionEntry= $this->getEntry($dbQuestionCuePoint->getEntryId(), $dbQuestionCuePoint->getType());
+			$this->explanation = $kQuestionEntry->explanation;
+
 			$dbUserEntry = UserEntryPeer::retrieveByPK($this->quizUserEntryId);
-			if ($dbUserEntry && $dbUserEntry->getStatus() == QuizPlugin::getCoreValue('UserEntryStatus', QuizUserEntryStatus::QUIZ_SUBMITTED))
+
+			if ($kQuiz->getShowCorrectKeyOnAnswer() )
 			{
-				if (!$kQuiz->getShowCorrectAfterSubmission())
+				$optionalAnswers = $dbQuestionCuePoint->getFromCustomData(QuestionCuePoint::CUSTOM_DATA_OPTIONAL_ANSWERS, null, array());
+				if(is_array($optionalAnswers))
 				{
-					$this->isCorrect = KalturaNullableBoolean::NULL_VALUE;
-					$this->correctAnswerKeys = null;
-					$this->explanation = null;
+					$optionalAnswers = KalturaOptionalAnswersArray::fromDbArray($dbQuestionCuePoint->getOptionalAnswers());
+				}
+
+				foreach ( $optionalAnswers as $optAnswer ) {
+					if ($optAnswer->isCorrect)
+					{
+						if (is_null($this->correctAnswerKeys))
+						{
+							$this->correctAnswerKeys = new KalturaOptionalAnswer();
+						}
+						$this->correctAnswerKeys->fromObject($optAnswer, $responseProfile);
+					}
 				}
 			}
-			else
-			{
-				if (!$kQuiz->getShowResultOnAnswer())
-					$this->isCorrect = KalturaNullableBoolean::NULL_VALUE;
 
-				if (!$kQuiz->getShowCorrectKeyOnAnswer())
+			if ($kQuiz->getShowResultOnAnswer())
+			{
+				$optionalAnswers = $dbQuestionCuePoint->getFromCustomData(QuestionCuePoint::CUSTOM_DATA_OPTIONAL_ANSWERS, null, array());
+				if(is_array($optionalAnswers))
+				{
+					$optionalAnswers = KalturaOptionalAnswersArray::fromDbArray($dbQuestionCuePoint->getOptionalAnswers());
+				}
+				/**
+				 * @var kOptionalAnswer $optAns
+				 */
+				foreach ( $optionalAnswers as $optAnswer )
 				{
-					$this->correctAnswerKeys = null;
-					$this->explanation = null;
+					if ($optAnswer->isCorrect && $this->answerKey == $optAnswer->key)
+					{
+						$this->isCorrect = true;
+						break;
+					}
+					else
```

Comment:
Instead of using else, set the value of isCorrect once before the loop to false and remove this else.